### PR TITLE
testing: add function `RelayPacketWithResults`

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #2956 

### Commit Message / Changelog Entry

```bash
test: add function `RelayPacketWithResults`
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: testing/path.go [hunk 1]
```diff
@@ -83,3 +85,54 @@ func (path *Path) RelayPacket(packet channeltypes.Packet) error {
...
 	return fmt.Errorf("packet commitment does not exist on either endpoint for provided packet")
 }
+
+// RelayPacketWithResults attempts to relay the packet first on EndpointA and then on EndpointB
+// if EndpointA does not contain a packet commitment for that packet. The function returns:
+// - The result of the packet receive transaction.
+// - The acknowledgement written on the receiving chain.
+// - An error if a relay step fails or the packet commitment does not exist on either endpoint.
+func (path *Path) RelayPacketWithResults(packet channeltypes.Packet) (res *sdk.Result, ack []byte, err error) {
+	pc := path.EndpointA.Chain.App.GetIBCKeeper().ChannelKeeper.GetPacketCommitment(path.EndpointA.Chain.GetContext(), packet.GetSourcePort(), packet.GetSourceChannel(), packet.GetSequence())
```
Comment (DimitrisJim) [inline testing/path.go:95]: to reduce duplication, we can now write `RelayPacket` as a wrapper of this which just drops the res, ack returns no?

**Review by DimitrisJim — APPROVED** [review]
leaving a small suggestion for reducing duplicate code, other than that lgtm!

File: testing/path.go (hunk 1 above)
Comment (crodriguezvega) [reply testing/path.go:95]: yes, great idea!

File: testing/path.go [hunk 2]
```diff
@@ -39,47 +41,57 @@ func (path *Path) SetChannelOrdered() {
...
+	_, _, err := path.RelayPacketWithResults(packet)
+	return err
+}
+
+// RelayPacketWithResults attempts to relay the packet first on EndpointA and then on EndpointB
+// if EndpointA does not contain a packet commitment for that packet. The function returns:
+// - The result of the packet receive transaction.
+// - The acknowledgement written on the receiving chain.
+// - An error if a relay step fails or the packet commitment does not exist on either endpoint.
+func (path *Path) RelayPacketWithResults(packet channeltypes.Packet) (res *sdk.Result, ack []byte, err error) {
```
Comment (chatton) [inline testing/path.go:53]: I don't think we should use named returns, I think these are pretty error prone. The name of the variables in the method body should be clear enough to indicate their purpose.

**Review by chatton — COMMENTED** [review]
Looks great! One suggestion to make the code a little less error prone!

**Review by damiannolan — APPROVED** [review]
LGTM!

File: testing/path.go [hunk 3]
```diff
@@ -39,47 +41,57 @@ func (path *Path) SetChannelOrdered() {
...
 		res, err := path.EndpointA.RecvPacketWithResult(packet)
 		if err != nil {
-			return err
+			return nil, nil, err
 		}
 
 		ack, err := ParseAckFromEvents(res.GetEvents())
 		if err != nil {
-			return err
+			return res, ack, err
```
Comment (colin-axner) [inline testing/path.go:90]: this should probably be `nil, nil` since we are returning an error?

File: testing/path.go [hunk 4]
```diff
@@ -39,47 +41,57 @@ func (path *Path) SetChannelOrdered() {
...
 		res, err := path.EndpointB.RecvPacketWithResult(packet)
 		if err != nil {
-			return err
+			return nil, nil, err
 		}
 
 		ack, err := ParseAckFromEvents(res.GetEvents())
 		if err != nil {
-			return err
+			return res, ack, err
```
Comment (colin-axner) [inline testing/path.go:69]: ditto

File: testing/path.go [hunk 5]
```diff
@@ -39,47 +41,57 @@ func (path *Path) SetChannelOrdered() {
...
 		}
 
-		return path.EndpointA.AcknowledgePacket(packet, ack)
+		return res, ack, path.EndpointA.AcknowledgePacket(packet, ack)
```
Comment (colin-axner) [inline testing/path.go:72]: I would prefer to split this up now that we disabled the linter which changed this (same for below)
```go
if err := path.EndpointA.AcknowledgePacket(packet, ack); err != nil {
    return nil, nil, err
}

return res, ack, nil
```

**Review by chatton — APPROVED** [review]

**Pull request merged**